sui_programmability/examples/nfts/sources/DiscountCoupon.move

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module NFTs::DiscountCoupon {
+    use Sui::Coin;
+    use Sui::SUI::SUI;
+    use Sui::NFT::{Self, NFT};
+    use Sui::TxContext::{Self, TxContext};
+
+    /// Discount coupon NFT.
+    struct DiscountCoupon has store {
```

Comment:
Is the plan to always store it as NFT? As an option we could add `store`. @sblackshear previously stopped me from wrapping things into NFTs (as Move objects are usually NFTs by their nature).

Reply:
I'm a bit confused by the comment here. It looks like this needs `store` to be wrapped in a `NFT<T>`. Is there something else I'm missing? If so, could you explain the background a bit :) 

(Side note, really disappointed y'all didn't name it `NF<T>` :P )

Reply:
You are totally right that NFT requires store, but we were (iirc) considering using structs directly or allowing `key` capability not to lock people inside NFTs.

Reply:
Having discussed with @sblackshear, @damirka and others, we might retire `NFT<T>` (or `NF<T>` per @tnowacki) soon. Other examples, including the naive [`Num` nft ](https://github.com/MystenLabs/sui/blob/main/sui_programmability/examples/nfts/sources/Num.move) use this pattern atm.

Reply:
Yep, I think we can/should get rid of this. Bad idea on my part + apologies!